src/modeler/AutoRest.Swagger/Validation/ArmResourcePropertiesBag.cs

```diff
@@ -49,7 +49,7 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<
             foreach (var violatingModel in violatingModels)
             {
                 var violatingProperties = definitions[violatingModel].Properties.Keys.Union(definitions[violatingModel].Properties["properties"].Properties.Keys).Intersect(ArmPropertiesBag);
-                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path), this, violatingModel, 
+                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty(violatingModel).AppendProperty("properties")), this, violatingModel, 
```

Comment:
not related to this change, but the comment for "GetValidationMessages" seems not to represent what this does, it'd be great to fix it as you're modifying this file.